### PR TITLE
Clean up code

### PR DESCRIPTION
**Pull request by wemeetagain**
- Many comments and variable alignment with sha256 spec:
 https://web.archive.org/web/20130526224224/http://csrc.nist.gov/groups/STM/cavp/documents/shs/sha256-384-512.pdf
- `buffer` -> `M` (message blocks)
- `temp` -> `W` (extended message blocks)
- `out` -> `output`
- declare `a`-`h`, `t1`,`t2`, `i` as global variables
- add `load32Bswap` inline function
- simplify `hashBlocks` to do a single round of the "main loop"
- removed `finished` variable / error throwing
- add test vector from the spec

### CODE REVIEW

File: assembly/index.ts [hunk 1]
```diff
@@ -51,73 +77,95 @@ function load8(ptr: usize, offset: usize): u8 {
...
+  d = H3;
+  e = H4;
+  f = H5;
+  g = H6;
+  h = H7;
+
+  // Load message blocks into first 16 expanded message blocks
+  for (i = 0; i < 16; i++) {
+    store32(wPtr, i,
+      load32Bswap(mPtr, i)
```
Comment (MaxGraey) [inline assembly/index.ts]: how bout rename `load32Bswap` to `load32be` (big endian)?

**Comment by MaxGraey** [comment]
I see some structural changes in `update` and `finish`. Could you run benchmark for compare old approach and new one?

**Comment by wemeetagain** [comment]
Looks roughly the same:

before:
```
input length 32 x 158,559 ops/sec ±0.66% (94 runs sampled)
input length 64 x 90,829 ops/sec ±0.74% (96 runs sampled)
input length 128 x 49,317 ops/sec ±0.87% (91 runs sampled)
input length 256 x 25,404 ops/sec ±1.06% (96 runs sampled)
input length 512 x 13,398 ops/sec ±0.80% (95 runs sampled)
```

after:
```
input length 32 x 156,400 ops/sec ±1.13% (91 runs sampled)
input length 64 x 89,539 ops/sec ±0.82% (98 runs sampled)
input length 128 x 48,774 ops/sec ±1.00% (93 runs sampled)
input length 256 x 25,818 ops/sec ±1.09% (95 runs sampled)
input length 512 x 13,394 ops/sec ±0.77% (96 runs sampled)
```

File: assembly/index.ts (hunk 1 above)
Comment (wemeetagain) [reply assembly/index.ts]: :+1:

**Comment by MaxGraey** [comment]
Great. Last thing. Could you also update to AS 0.9.2 ?

**Pull request merged**